src/eth/TransactionManager.js

```diff
@@ -210,6 +210,23 @@ class Tracker {
     return txs[0];
   }
 
+  clearExpiredListeners() {
+    each(this._transactions, (val, key) => {
+      val.forEach(tx => {
+        // TODO: check Tx age for 5 minutes or older:
+        // const age = (new Date() - new Date(tx._timeStampMined)) / 1000;
+        if (tx.state() === 'finalized' || tx.state() === 'error') {
```

Comment:
there are convenience methods `isFinalized` and `isError` you can use, from `TransactionLifecycle`